### ISSUE TITLE
Universally loaded images

### ISSUE THREAD

**Issue by dferber90** [dferber90/webapp#5]
To embed an image it should be placed in `public` folder. Then, it can be loaded using a webpack loader:

``` js
// webpack.config.js
var imageLoader = {
  test: /\.(png|jpg)$/,
  loader: 'url-loader?limit=8192',
  exclude: /node_modules/
}
```

And in the file where it is used, it is imported with:

``` js
import someImg from './public/someImg.png'
```

This will put the path of `someImg.png` into the variable `someImg`. In case the image is small, it will put a data-uri string as the value of `someImg`.

In case the js-file importing the image is universal (used by the client and the server), it will be loaded twice by webpack. Once for the client and once for the server configuration. The client will generate `build/public/assets/0845a58fef22e11981898cfb8d7ed119.png` which is correct. Since the server outputs files directly to `build` it will be something like `build/0845a58fef22e11981898cfb8d7ed119.png`, which differs from the client and is wrong.  Now, we need some way to let the server "load" the file to get its correct final path, without actually generating it. The mismatch of paths happens because `output.path` currently differs for the client and the server.

**why the server can't simply ignore images**
If the server would ignore images with the IgnorePlugin, the server would not be able to get their paths or data-uri strings. It must be able to read them, but it should not emit the files. The client should do that.

**what about images the server imports, but the client does not import**
These images would be missing in case the server does not emit any files.

**Comment by dferber90** [comment]
This is a very entangled situation. Not sure about the best way. I have a proof of concept where the client and the server both emit image files, potentially doing the work twice.

To get this to work, the `output.path`s have to be equal. But this causes `backend.js` to be generated into `build/public/assets/`, because the images have to go into this directory as well. One option is to move `backend.js` moved automatically back into `build/` after webpack is done.

Maybe something is fundamentally wrong about the configured output paths, that's causing this.

**Comment by dferber90** [comment]
Solved with 19557ae2f93f7a345e1504223a0d6f309211ad5b. The solution was to add a relative path to `output.filename` that causes `backend.js` to go into `build/`.

``` js
    output: {
      path: outputPath, // put images into build/public/assets
      filename: '../../backend.js', // put backend.js into 'build/backend.js'
      publicPath: '/assets/'
    },
```

Now none of that manual moving of `backend.js` has to be done.

Downside: Images are still generated twice.
